recipes-bsp/u-boot/u-boot-artik.bb

```diff
@@ -21,3 +21,33 @@ PACKAGE_ARCH = "${MACHINE_ARCH}"
 COMPATIBLE_MACHINE = "(artik5|artik10)"
 
 RDEPENDS_${PN} = "secure-boot-artik"
+
+do_compile_append() {
+    cp `find . -name "env_common.o"` copy_env_common.o
+    ${OBJCOPY} -O binary --only-section=.rodata `find . -name "copy_env_common.o"`
+    tr '\0' '\n' < copy_env_common.o | grep '=' | tee default_envs_emmc.txt default_envs_sd.txt > /dev/null
+
+    # our root partition is number 2 (exit with code 1 when no replacement done so we error out when upstream modifies the u-boot env)
+    sed -i '/rootpart=[0-9]\{1\}/{s//rootpart=2/;h};${x;/./{x;q0};x;q1}' default_envs_emmc.txt
+    sed -i '/rootpart=[0-9]\{1\}/{s//rootpart=2/;h};${x;/./{x;q0};x;q1}' default_envs_sd.txt
```

Comment:
Could you redo this part with a variable defined in this file, for example ROOTPART ?= "2" and use it here in this sed? This way, other layers needing to change the root partition would just redefine this ROOTPART variable in the u-boot-artik.bbappend.